BUG: Fix timedelta64+Timestamp, closes #24775

jbrockmendel opened the pull request:
- [x] closes #24775
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

jbrockmendel committed:
BUG: Fix timedelta64+Timestamp, closes #24775

jreback commented on pandas/tests/scalar/timestamp/test_arithmetic.py:
```diff
@@ -112,3 +112,9 @@ def test_addition_subtraction_preserve_frequency(self):
         assert (ts + td64).freq == original_freq
         assert (ts - td64).freq == original_freq
+
+    def test_radd_timedelta64(self):
```
seems like a lone test; can u  co-locate with same for Timedelta and (or parametrize)

jbrockmendel replied:
This is definitely the appropriate place.  I think there are other tests in tests.scalar.timestamp that belong in this file

jreback replied:
my point is to also include pd.Timedelta here as well via parameterization. I suspect we *already* test this elsewhere, so rather than add a one-off, should add elsewhere (or simply combine / move them here if this is the final destination).

```
(pandas) bash-3.2$ grep -r _radd pandas/tests/ --include '*.py'
pandas/tests//tseries/offsets/test_offsets.py:    def test_radd(self):
pandas/tests//frame/test_query_eval.py:            for op_str, op, rop in [('+', '__add__', '__radd__'),
pandas/tests//arithmetic/test_datetime64.py:        if op_str not in ['__add__', '__radd__', '__sub__']:
pandas/tests//arithmetic/test_datetime64.py:        if op_str not in ['__add__', '__radd__', '__sub__', '__rsub__']:
pandas/tests//arithmetic/test_datetime64.py:    @pytest.mark.parametrize('op', ['__add__', '__radd__',
pandas/tests//arithmetic/test_timedelta64.py:    # Tests for timedelta64[ns] __add__, __sub__, __radd__, __rsub__
pandas/tests//arithmetic/test_numeric.py:    # __add__, __sub__, __radd__, __rsub__, __iadd__, __isub__
pandas/tests//arithmetic/test_numeric.py:    def test_series_frame_radd_bug(self):
pandas/tests//arithmetic/test_object.py:    def test_objarr_radd_str(self, box):
pandas/tests//arithmetic/test_object.py:    def test_objarr_radd_str_invalid(self, dtype, data, box):
pandas/tests//arithmetic/test_object.py:    def test_series_with_dtype_radd_timedelta(self, dtype):
pandas/tests//scalar/timedelta/test_arithmetic.py:        __add__, __radd__,
pandas/tests//scalar/timedelta/test_arithmetic.py:            # datetime + Timedelta does _not_ call Timedelta.__radd__,
pandas/tests//indexes/test_category.py:        (lambda idx: ['a', 'b'] + idx, '__radd__'),
```

jbrockmendel replied:
> I suspect we already test this elsewhere, so rather than add a one-off, should add elsewhere (or simply combine / move them here if this is the final destination).

well we don't already test this specifically, since this test fails on master.  Most of the grep results posted are for non-scalars; we definitely don't want to start mixing the scalar tests in with those.  I'll collect other Timestamp arithmetic tests that are currently misplaced and put them in this file.

jreback replied:
> > I suspect we already test this elsewhere, so rather than add a one-off, should add elsewhere (or simply combine / move them here if this is the final destination).
> 
> well we don't already test this specifically, since this test fails on master. Most of the grep results posted are for non-scalars; we definitely don't want to start mixing the scalar tests in with those. I'll collect other Timestamp arithmetic tests that are currently misplaced and put them in this file.

If you read my point is that we certaily test this for ``Timedelta``. so these belong together.

jschendel commented on doc/source/whatsnew/v0.25.0.rst:
```diff
@@ -567,6 +567,7 @@ Datetimelike
 - Bug in :meth:`isin` for datetimelike indexes; :class:`DatetimeIndex`, :class:`TimedeltaIndex` and :class:`PeriodIndex` where the ``levels`` parameter was ignored. (:issue:`26675`)
 - Bug in :func:`to_datetime` which raises ``TypeError`` for ``format='%Y%m%d'`` when called for invalid integer dates with length >= 6 digits with ``errors='ignore'``
 - Bug when comparing a :class:`PeriodIndex` against a zero-dimensional numpy array (:issue:`26689`)
+- Bug where adding :class:`Timestamp` to a ``np.timedelta64`` object would raise instead of returning a :class:`Timestamp` (:issue:24775)
```
missing backticks around the issue number

jbrockmendel committed:
fix missing backticks

jbrockmendel committed:
fix comparison bug

jreback commented on pandas/tests/scalar/timestamp/test_comparisons.py:
```diff
@@ -156,6 +156,18 @@ def test_timestamp_compare_with_early_datetime(self):
+    def test_compare_zerodim_array(self):
+        # GH#26916
+        ts = Timestamp.now()
+        td64 = np.datetime64('2016-01-01', 'ns')
```
this is odd named td64 no?

jbrockmendel replied:
typo, will update

jbrockmendel committed:
Collect Timestamp arith tests, parametrize

jbrockmendel commented:
Fixed typo, determined there's only a couple other dedicated Timestamp arith tests, moved them over to the appropriate place and parametrized them.

We've done a good job at making tslibs self-contained, less so with the tests.  At some point I'd like to change that, hopefully make it easier to identify gaps in isolated coverage.

jreback commented:
> Fixed typo, determined there's only a couple other dedicated Timestamp arith tests, moved them over to the appropriate place and parametrized them.
> 
> We've done a good job at making tslibs self-contained, less so with the tests. At some point I'd like to change that, hopefully make it easier to identify gaps in isolated coverage.

yep i here ya increasing coverage always +1

jreback commented on pandas/tests/scalar/timestamp/test_arithmetic.py:
```diff
@@ -112,3 +112,21 @@ def test_addition_subtraction_preserve_frequency(self):
+        td = np.timedelta64(3, 'h')
+        assert td + ts == ts + td
+
+    @pytest.mark.parametrize('other,exdiff', [
```
can you do: exdiff -> expected_difference

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into tsadd

jbrockmendel committed:
rename exdiff-->expected_difference

TomAugspurger reviewed (approved):
This seems good. Can you fix the merge conflict and merge on green?

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into tsadd

jreback commented:
i have some open requests here

jbrockmendel commented:
> i have some open requests here

I think your open request is that we put this with the Timedelta arithmetic tests.  But doing so would break the organization of the Timedelta tests.

jreback commented:
no my requests are to parametrize over a Timedelta the tests u added

jbrockmendel committed:
Merge branch 'master' of https://github.com/pandas-dev/pandas into tsadd

jbrockmendel committed:
parametrize test

jbrockmendel commented:
> no my requests are to parametrize over a Timedelta the tests u added

added.

jreback commented:
thanks @jbrockmendel
